dipy/viz/regtools.py

```diff
@@ -90,162 +93,171 @@ def draw_lattice_2d(nrows, ncols, delta):
+    warped_backward : array
+        Image with grid showing the effect of transforming the static image to
+        the moving image.  Shape will be `inverse_grid_shape` if specified,
+        otherwise the shape of the moving image.
+
     Note
     ----
     The default value for the affine transformation is "-1" to handle the case
     in which the user provides "None" as input meaning "identity". If we used
     None as default, we wouldn't know if the user specifically wants to use
     the identity (specifically passing None) or if it was left unspecified,
-    meaning to use the apropriate default matrix
-
+    meaning to use the apropriate default matrix.
     """
     if mapping.is_inverse:
-        #By default, direct_grid_shape is the codomain grid
+        # By default, direct_grid_shape is the codomain grid
         if direct_grid_shape is None:
             direct_grid_shape = mapping.codomain_shape
         if direct_grid_affine is -1:
             direct_grid_affine = mapping.codomain_affine
 
-        #By default, the inverse grid is the domain grid
+        # By default, the inverse grid is the domain grid
         if inverse_grid_shape is None:
             inverse_grid_shape = mapping.domain_shape
         if inverse_grid_affine is -1:
             inverse_grid_affine = mapping.domain_affine
     else:
-        #Now by default, direct_grid_shape is the mapping's input grid
+        # Now by default, direct_grid_shape is the mapping's input grid
         if direct_grid_shape is None:
             direct_grid_shape = mapping.domain_shape
         if direct_grid_affine is -1:
             direct_grid_affine = mapping.domain_affine
 
-        #By default, the output grid is the mapping's domain grid
+        # By default, the output grid is the mapping's domain grid
         if inverse_grid_shape is None:
             inverse_grid_shape = mapping.codomain_shape
         if inverse_grid_affine is -1:
             inverse_grid_affine = mapping.codomain_affine
 
-    #The world-to-image (image = drawn lattice on the output grid)
-    #transformation is the inverse of the output affine
+    # The world-to-image (image = drawn lattice on the output grid)
+    # transformation is the inverse of the output affine
     world_to_image = None
     if inverse_grid_affine is not None:
-        world_to_image = np.linalg.inv(inverse_grid_affine) 
+        world_to_image = np.linalg.inv(inverse_grid_affine)
 
-    #Draw the squares on the output grid
-    X1,X0 = np.mgrid[0:inverse_grid_shape[0], 0:inverse_grid_shape[1]]
-    lattice_out=draw_lattice_2d((inverse_grid_shape[0] + delta) / (delta + 1), 
-                                (inverse_grid_shape[1] + delta) / (delta + 1),
-                                delta)
-    lattice_out=lattice_out[0:inverse_grid_shape[0], 0:inverse_grid_shape[1]]
+    # Draw the squares on the output grid
+    lattice_out = draw_lattice_2d(
+        (inverse_grid_shape[0] + delta) // (delta + 1),
```

Comment:
Hi @matthew-brett, =)
thank you very much for reviewing this!, this is correct, it is actually integer division rounding up: (x+(n-1))//n is division by n rounding up. This considers the case delta=0 (the degenerate case where there is no separation between division lines).